src/databricks/labs/lsql/core.py

```diff
@@ -70,28 +72,37 @@ def __new__(cls, *args, **kwargs):
         return tuple.__new__(cls, args)
 
     @classmethod
-    def factory(cls, col_names: list[str]) -> type:
+    def factory(cls, col_names: list[str]) -> type["Row"]:
         """Create a new Row class with the given column names."""
         return type("Row", (Row,), {"__columns__": col_names})
 
-    def as_dict(self) -> dict[str, Any]:
-        """Convert the row to a dictionary with the same conventions as Databricks SDK."""
-        return dict(zip(self.__columns__, self, strict=True))
+    if sys.version_info >= (3, 13):
```

Comment:
Is this necessary? Or just to forward/backward compatible? Would the `warnings.warn not be sufficient?

Reply:
Good question: I'll clarify, and add a comment.

Traditionally deprecation in Python is handled by raising that warning. This is fine, but code analysis (type-checkers, IDE completion, etc) isn't able to determine that a method is deprecated and shouldn't be used because there's nothing on the interface signalling it.

This was recently improved (via [PEP 702](https://peps.python.org/pep-0702/)) with an annotation, `@warnings.deprecation` that both emits the traditional warning _and_ signals to the type system (and thus code analysis) that a method is deprecated. It's only available from 3.13 onwards, which is why this guard is here.

Including it here in this way means that IntelliJ recognises the method as deprecated and flags usage as such, which is immediately a win to help avoid accidentally using it.